src/network/core/game_info.cpp

```diff
@@ -36,45 +36,38 @@ NetworkServerGameInfo _network_game_info; ///< Information about our game.
 
 /**
  * Get the network version string used by this build.
- * The returned string is guaranteed to be at most NETWORK_REVISON_LENGTH bytes.
+ * The returned string is guaranteed to be at most NETWORK_REVISON_LENGTH bytes including '\0' terminator.
  */
 const char *GetNetworkRevisionString()
 {
-	/* This will be allocated on heap and never free'd, but only once so not a "real" leak. */
-	static char *network_revision = nullptr;
+	static std::string network_revision;
 
-	if (!network_revision) {
+	if (network_revision.empty()) {
 		/* Start by taking a chance on the full revision string. */
-		network_revision = stredup(_openttd_revision);
+		network_revision = _openttd_revision;
 		/* Ensure it's not longer than the packet buffer length. */
-		if (strlen(network_revision) >= NETWORK_REVISION_LENGTH) network_revision[NETWORK_REVISION_LENGTH - 1] = '\0';
+		if (network_revision.size() >= NETWORK_REVISION_LENGTH) network_revision.resize(NETWORK_REVISION_LENGTH - 1);
 
 		/* Tag names are not mangled further. */
-		if (_openttd_revision_tagged) {
-			Debug(net, 3, "Network revision name: {}", network_revision);
-			return network_revision;
+		if (!_openttd_revision_tagged) {
+			/* Prepare a prefix of the git hash. */
+			assert(_openttd_revision_modified < 3);
+			std::string githash_suffix = fmt::format("-{}{}", "gum"[_openttd_revision_modified], _openttd_revision_hash);
+			if (githash_suffix.size() > GITHASH_SUFFIX_LEN) githash_suffix.resize(GITHASH_SUFFIX_LEN);
+
+			/* Where did the hash start in the original string? Overwrite from that position, unless that would go past end of packet buffer length. */
+			size_t hash_end = network_revision.find_last_of('-');
```

Comment:
Where your version differs from the original is at this point. Say, I have a really long branch name:

`more_string_in_networdfsfsdk`
The `_openttd_revision` would become something like `20210613-more_string_in_networdfsfsdk-m34ed1eeee4`. This is cut to `20210613-more_string_in_networdfsfs` (or something, didn't really check, but without the `-`). And now the "hash" is removed, leaving `20210613-`. This means the result is `20210613-m34ed1eeee4` instead of `20210613-more_string-m34ed1eeee4`.

I am not 100% sure if `_openttd_revision` allows this to start with, but really long branch names are pretty common.

In your godbolt you can see this problem in action with the above string. Basically, you were missing the case where the second `-` was passed the `NETWORK_REVISION_LENGTH` :)